File: Medium/Python/240. Search a 2D Matrix II.py
class Solution(object):
    def searchMatrix(self, matrix, target):
        row = len(matrix)
        col = len(matrix[0])
        i = 0
        j = col - 1

        while i < row and 0 <= j:
            if matrix[i][j] == target:
                return True
            elif target < matrix[i][j]:
                j -= 1
            else:
                i += 1

        return False

File: Medium/JavaScript/240. Search a 2D Matrix II.js
var searchMatrix = function(matrix, target) {
    let row = matrix.length;
    let col = matrix[0].length;
    let i = 0;
    let j = col - 1;

    while(i < row && 0 <= j){
        if(matrix[i][j] == target)
            return true;
        else if(target < matrix[i][j])
            j--;
        else
            i++;
    }

    return false;
};
